src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -245,7 +245,8 @@ var ReactCompositeComponentMixin = {
     inst.props = publicProps;
     inst.context = publicContext;
     inst.refs = emptyObject;
-    inst.updater = ReactUpdateQueue;
+    inst.updater = transaction.getUpdateQueue ?
```

Comment:
Any transactions that might end up here would need this method. IIRC that's just Reconcile and ServerRendering transactions but I might be wrong.